src/main/java/com/backbase/ct/bbfuel/setup/CapabilitiesDataSetup.java

```diff
@@ -179,12 +188,34 @@ private void ingestActionsPerUser() {
 
     private void ingestBillPayUsers() {
         if (this.globalProperties.getBoolean(PROPERTY_INGEST_BILLPAY)) {
-            this.accessControlSetup.getLegalEntitiesWithUsersExcludingSupport()
-                .stream()
-                .map(LegalEntityWithUsers::getUserExternalIds)
-                .flatMap(List::stream)
-                .collect(Collectors.toList())
-                .forEach(this.billpayConfigurator::ingestBillPayUser);
+            this.accessControlSetup.getLegalEntitiesWithUsersExcludingSupport().forEach((le) -> {
+
+                List<Account> accounts = new ArrayList<>();
+                String externalUserId = le.getUserExternalIds().get(0);
+                loginRestClient.login(externalUserId, externalUserId);
+                String externalLegalEntityId = this.userPresentationRestClient.
+                    retrieveLegalEntityByExternalUserId(externalUserId)
+                    .getExternalId();
+                List<ArrangementItem> arrangementItems = accountsIntegrationRestClient
+                    .getArrangements(externalLegalEntityId);
+
+                arrangementItems.forEach((arrangement) -> {
+                    accounts.add(mapBillPayAccount(arrangement));
+                });
+                le.getUserExternalIds().forEach((user) -> {
+                    billpayConfigurator.ingestBillPayUser(user);
+                    if (this.globalProperties.getBoolean(PROPERTY_INGEST_BILLPAY_ACCOUNTS)) {
+                        billpayConfigurator.ingestBillPayAccounts(user, accounts);
+                    }
+                });
+            });
         }
     }
+
+    private Account mapBillPayAccount(ArrangementItem arrangement) {
+        return new Account()
+            .withAccountNumber(arrangement.getBBAN())
```

Comment:
why not use IBAN some of the times?

Reply:
This change has been requested to ensure that the account data in Bill Pay matches those in Product Summary. In Bill Pay, the account balance is retrieved from Product Summary via searching for the BBAN / bankBranchCode combination so this must always use the BBAN.